### ISSUE TITLE
Error is assigning weight decay

### ISSUE THREAD

**Issue by divyat09** [gregorbachmann/Next-Token-Failures#5]
The input weight decay via the argparse is not passed to the AdamW optimizer, rather it is hardcoded to zero. 

https://github.com/gregorbachmann/Next-Token-Failures/blob/main/train.py#L144
